lib/auth/auth.go

```diff
@@ -486,10 +508,93 @@ func (a *Server) runPeriodicOperations() {
+	// scrapped/replaced once the inventory and version-control systems
+	// are a bit further along.
+
+	var loadFailed bool
+	current := vc.Normalize(teleport.Version)
+
+	latest, err := github.LatestStable()
+	if err != nil {
+		log.Warnf("Failed to load github releases: %v (this will not impact teleport functionality)", err)
+		loadFailed = true
+		latest = current
+	}
+
+	// use visitor to find the oldest version among connected instances
+	// TODO(fspmarshall): replace this check as soon as we have a backend inventory repr. using
+	// connected instances is a poor approximation and may lead to missed notifications if auth
+	// server is up to date, but instances not connected to this auth need update.
+	var instanceVisitor vc.Visitor
+	a.inventory.Iter(func(handle inventory.UpstreamHandle) {
+		instanceVisitor.Visit(vc.Normalize(handle.Hello().Version))
+	})
+
+	msg := makeUpgradeSuggestionMsg(current, latest, instanceVisitor.Oldest())
+
+	if msg != "" {
+		alert, err := types.NewClusterAlert(
+			alertID,
+			msg,
+			types.WithAlertLabel(types.AlertOnLogin, "yes"),
+			types.WithAlertLabel(types.AlertPermitAll, "yes"),
+		)
+		if err != nil {
+			log.Warnf("Failed to build %s alert: %v (this is a bug)", alertID, err)
+			return
+		}
+		if err := a.UpsertClusterAlert(ctx, alert); err != nil {
+			log.Warnf("Failed to set %s alert: %v", alertID, err)
+			return
+		}
+	} else if !loadFailed {
+		log.Debugf("Cluster appears up to date, clearing %s alert.", alertID)
+		err := a.DeleteClusterAlert(ctx, alertID)
+		if err != nil && !trace.IsNotFound(err) {
+			log.Warnf("Failed to delete %s alert: %v", alertID, err)
 		}
 	}
 }
 
+// makeUpgradeSuggestionMsg generates an upgrade suggestion alert msg if one is
+// needed (returns "" if everything looks up to date).
+func makeUpgradeSuggestionMsg(current, latest, oldestInstance string) string {
+	// check if this teleport instance wants upgrade
+	if semver.Compare(latest, current) == 1 {
+		// specialize the message a bit to distinguish between a major and minor upgrade.
+		var msg string
+		if semver.Major(latest) != semver.Major(current) {
+			log.Debugf("Generating alert msg for new major version. current=%s, latest=%s", current, latest)
+			msg = fmt.Sprintf("A new major version of Teleport is available. Please consider upgrading your cluster to %s.", semver.Major(latest))
+		} else {
+			log.Debugf("Generating alert msg for new minor or patch release. current=%s, latest=%s", current, latest)
+			msg = fmt.Sprintf("Teleport %s is now available, please consider upgrading your cluster.", latest)
+		}
+		return msg
+	}
+
+	if oldestInstance != "" && semver.Compare(latest, oldestInstance) == 1 {
+		log.Debugf("Generating alert msg for older peripheral instance(s). latest=%s, oldestInstance=%s", latest, oldestInstance)
+		// at least one connected instance is older than this auth server.
+		return "Cluster contains some older Teleport instances. Please consider upgrading."
```

Comment:
Have we standardized on any nouns? Specifically:

- What do we want to call the Teleport Cluster? Cluster, Tenant, Control Plane, Auth/Proxy, or something else?
- What do we want to call Teleport agents? Agents, services, instances, etc.?

I think for the time being, the following is more descriptive:

```suggestion
Some resources connected to this Teleport Cluster are running older versions of the Teleport Service. Please consider upgrading your resources to run the latest version: <link to latest version here>. 
```

Reply:
I think "service" has a well established meaning, and usually refers to an abstract entity, many of which may be running with a single teleport process ('Auth Service', 'SSH Service', etc).  This is the model used everywhere in the code, and is reflected in our config files and user-facing messages.

Currently, when speaking about the unique identity held by teleport running on a given server, we use the term `Instance` in the code (e.g. instance `foo` runs services `Auth` and `Node`).

I'm a bit unsure how we define "Agent".  I've heard it used in ways that seem to indicate a "peripheral process not running auth/proxy", but I've also heard it used as broadly as referring to multiple copies of teleport with the same identity (such as can happen during graceful restarts) as being individual agents.  Either interpretation would make "Agent" suitable for this alert message though.

I'm also wary of using the term "resource" for a teleport installation/instance/agent/etc.  We use the term "resource" so extensively in the teleport codebase to specifically refer to _configuration resources_, that I see this usage as a potential source of confusion.

Given the above, how about limiting the vocabulary to cluster/agent in these message?.  Maybe something like this:

```
Some Agents within this Cluster are running an older version of the Teleport.  Please consider upgrading them. [<link>]
```